pkg/odo/genericclioptions/context.go

```diff
@@ -72,6 +72,59 @@ func checkProjectCreateOrDeleteOnlyOnInvalidNamespace(command *cobra.Command, er
 	}
 }
 
+// getFirstChildOfCommand gets the first child command of the root command of command
+func getFirstChildOfCommand(command *cobra.Command) *cobra.Command {
+	// If command does not have a parent no point checking
+	if command.HasParent() {
+		// Get the root command and set current command and its parent
+		r := command.Root()
+		p := command.Parent()
+		c := command
+		for {
+			// if parent is root, then we have our first child in c
+			if p == r {
+				return c
+			}
+			// Traverse backwards making current command as the parent and parent as the grandparent
+			c = p
+			p = c.Parent()
+		}
+	}
+	return nil
+}
+
+func checkValidConfig(command *cobra.Command, configFileName string) {
+	_, err := config.NewLocalConfigInfo(configFileName, true)
```

Comment:
NewLocalConfigInfo actually reads from file everytime its invoked
I believe the aim of this PR is also to decrease the number of times its called which is the reason I believe why we are moving config resolution to context
In my opinion component settings is parallel to context except for only one field in context which is client so should Context struct also be made to contain LocalConfigInfo? and remove project, component name and app from context bcoz they are there in LocalConfigInfo

Reply:
in which case there probably is no need to resolve each of project, component and app in my opinion bcoz now the single source of truth in my opinion should be config

Reply:
Unless overridden by respective parameters :)

Reply:
https://github.com/openshift/odo/issues/1520#issuecomment-477943538

No flags related to component or app needs to be passed to any command other than component create, if passed they'll be ignored or will error out

Reply:
Cool then

Reply:
Updated. i really feel like removing those flags right now. Already have it in my git stash :)